### ISSUE TITLE
Invalidate CF distribution on deploy of static site

### ISSUE THREAD

**Issue by yannickvr** [aws/copilot-cli#5024]
The Static Site Service does currently not invalidate existing assets on the cloudfront distribution when redeploying the service, so e.g. when index.html is updated to add a new menu item, it's not immediately visible. 

I think there should be an invalidations option for the static site service, e.g.

```
name: static-site
type: Static Site
files:
  - source: source
    recursive: true
    invalidations:
        - index.html

**Comment by huanjani** [comment]
Hi, @yannickvr. 
Thanks for the feedback! We're actually working on cache validation currently. Our plan is to invalidate any cached files upon deployment by default. Do you feel strongly that users should have the ability to specify files for invalidation and that all files shouldn't be invalidated by default?

**Comment by yannickvr** [comment]
> Hi, @yannickvr. Thanks for the feedback! We're actually working on cache validation currently. Our plan is to invalidate any cached files upon deployment by default. Do you feel strongly that users should have the ability to specify files for invalidation and that all files shouldn't be invalidated by default?

I would argue that invalidating all by default would be good standard behavior, but an option to exclude/include certain files/folders could be high on the wishlist. E.g. I would normally be fine with my `/pictures` folder not validating every deployment, as it could contain thousands of pictures that are normally only added and never changed, but my earlier example of  indexes etc. I would want to do every deploy, as they would change over their livecycle

**Comment by iamhopaul123** [comment]
Hello @yannickvr. 

> but an option to exclude/include certain files/folders could be high on the wishlist.

Basically if we support doing invalidation by default after each deployment, you would want to exclude certain files/folders because very likely they won't change and doing invalidation for them will lead to extra invalidation time?

**Comment by yannickvr** [comment]
> Hello @yannickvr.
> 
> > but an option to exclude/include certain files/folders could be high on the wishlist.
> 
> Basically if we support doing invalidation by default after each deployment, you would want to exclude certain files/folders because very likely they won't change and doing invalidation for them will lead to extra invalidation time?

Exactly, though doing some research, I'm not completely sure if that makes sense. It seems that a wildcard invalidation `/*` would count as a single path and be billed as such (1000 invalidations free/mo and then $0.005 per invalidation), that would suggest a wildcard invalidation would be faster then invalidating specific paths as well.

**Comment by huanjani** [comment]
This enhancement has now been released in v1.29.0: https://github.com/aws/copilot-cli/releases/tag/v1.29.0!
